lib/index.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const react_1 = require("react");
+const HOOK_PUBLISH_INDEX = 1;
+const getUUID = (a, b) => {
+    for (b = a = ''; a++ < 36; b += 4 << ~a * 6.5 ? (a ^ 15 ? 8 ^ Math.random() * (a ^ 20 ? 16 : 4) : 4).toString(16) : '-')
+        ;
+    return b;
+};
+const topics = {};
+const getCurrentState = (topic) => {
+    const topicRecord = topics[topic];
+    if (topicRecord && topicRecord.hasBeenPublished) {
+        return topicRecord.currentState;
+    }
+    else {
+        return undefined;
+    }
+};
+const publish = (topic) => (newState) => {
+    const topicRecord = topics[topic];
+    if (topicRecord) {
+        Object.values(topicRecord.subscriptionMap).forEach(publicHook => publicHook[HOOK_PUBLISH_INDEX](newState));
+        topicRecord.currentState = newState;
+        topicRecord.hasBeenPublished = true;
+    }
+    else {
+        throw new Error(`Cannot publish to non-existent topic "${topic}"`);
+    }
+};
+const unsubscribe = (topic, subscriptionId) => () => {
+    const topicRecord = topics[topic];
+    if (topicRecord && subscriptionId) {
+        delete topicRecord.subscriptionMap[subscriptionId];
+    }
+};
+function usePubSub(topic, defaultState) {
+    if (arguments.length !== 2) {
+        throw new Error(`Invalid hook usage; hook must be initialized with two arguments. The first
```

Comment:
Ah got it. I guess throwing an error here is the only way we can enforce args since we don't have compile-time checking without TS?

Reply:
Would it make sense to also then check for types on the args? Maybe at least make sure topic is a string?

Reply:
Haha nvm this is the transpiled JS file :P